jobserver/github.py

```diff
@@ -168,6 +184,57 @@ def create_issue(self, org, repo, title, body, labels):
 
         return r.json()
 
+    def create_repo(self, org, repo):
+        path_segments = [
+            "repos",
+            "opensafely",
+            "research-template",
+            "generate",
+        ]
```

Comment:
Hmm, so I'm not sure about using the research-template for interactive, as it's setup with base files we don't want, and a README that's misleading in this case. Also, it runs an initial action to render the repo template, which given we're automating things could be a bit messy.

I think it might be better to create a blank repo, and include a standard boiler plate README.md in every commit that spells out that this was a repo for an Interactive Project, with links to the project/workspace on jobs.opensafely.org.

Reply:
Alternatively, maybe an alternate interactive specific template might be better? And you can pass the template in as an arg to create_repo?

Reply:
The current research-template's README is already a boilerplate!  I'm not sure we need anything in there about interactive?

A separate template might well work but I'm leaning towards not doing that. We're going to wipe the working directory each time we generate code anyway.  Also it saves maintaining another repo!

I've added the ability to pass the template org/repo and renamed the method slightly to make it clearer what we're doing under the hood.

Reply:
Yes, but it's boilerplate for a very different thing - a manually authored study.

I think the interactive repos should have a README something like:

```
# OpenSAFELY Interactive Repository for Project Foo

This repo contains the code for the [OpenSAFELY Interactive] study for [project Foo].  Unlike regular studies, this code is not hand written, and is instead automatically generated from [a template], 

Any published results from this study will be available here: ...
```

This could possibly be rendered on every commit we do, and is not integral to walking skeleton, so we can defer decision.

But my other concern I think make cause issues in tests.

When you clone a repo from the research-template, it runs an initial action that updates the README template and then deletes itself. This takes a minute or two to run.

It's entire possible, especially in automated testing, that we a) create repo b) clone c) initial setup job finishes, making our clone out of date b) we commit/push, and thus error.

Reply:
Adding it to the analysis template repo for now sounds fine but I'm keen to only have one readme template if possible, that was a big point of the work when changing it for repo sign-offs!  I have a feeling we can handle the differences on job-server rather than in repo READMEs _but_ we should discuss this with the OSI team too (I'll drop a note in channel).

Reply:
Wouldn't making it part of the template repo mean it was the same for all actions…?  My goal was to have it all the same so now I'm a bit confused why having it in the template directory with nothing changed would not achieve that…!

Reply:
Not as currently implemented no, as each templated analysis is separate.

Additionally, we'd need add additional data in the template context.

I think this belongs in InteractiveAnalysisTemplate - its a step analogous to downloading a codelist, the same process for rendering all templated analysis.

Reply:
Switched to creating a repo without the research template as discussed 👍